### PR TITLE
Added zookeeper host configuration. Closes #1960

### LINKED ISSUE #1960: Configure sandbox binding
**FEATURE REQUEST**
When running the docker sandbox, the default namespace for DistributedLog is bonded to the 127.0.0.1.

I think it would be nice to configure such IP, as for development sometimes there are multiple virtual machines and containers running. And for fast prototyping, the sandbox is very useful, but when deploying code on a VM that same machine will not be able to connect correctly to the sandbox.

Meanwhile, a new namespace with the correct IP can be created to workaround.

### PR DESCRIPTION
**Pull request by hugomiguelabreu**
Descriptions of the changes in this PR:
Added host IP binding to sandbox.

### Motivation
Allowing running the sandbox inside guest machines and be accessible by all system (including other guest machines). When binding 127.0.0.1 other guest machines will experience an error. By allowing to bind the correct IP the namespace will be available and correctly configured for other hosts.

### Changes
Added a new argument to the _dlog_ tool allowing to define the host IP to bind ZooKeeper and DistributedLog namespace.

Master Issue: #1960

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [x] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [x] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [x] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java11]: skip build on java11. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [x] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [x] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [x] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

File: stream/distributedlog/core/src/main/java/org/apache/distributedlog/LocalDLMEmulator.java [hunk 1]
```diff
@@ -308,13 +308,15 @@ public static URI createDLMURI(String zkServers, String path) throws Exception {
...
         try {
             if (args.length < 1) {
-                System.out.println("Usage: LocalDLEmulator <zk_port>");
+                System.out.println("Usage: LocalDLEmulator <zk_host> <zk_port>");
                 System.exit(-1);
             }
 
-            final int zkPort = Integer.parseInt(args[0]);
+            final String zkHost = args[0];
+            final int zkPort = Integer.parseInt(args[1]);
```
Comment (eolivelli) [inline stream/distributedlog/core/src/main/java/org/apache/distributedlog/LocalDLMEmulator.java]: Can we make it compatible with only version ?
Just handle the case in which you have only one parameter

Comment (hugomiguelabreu) [reply stream/distributedlog/core/src/main/java/org/apache/distributedlog/LocalDLMEmulator.java]: Done 😄

**Comment by sijie** [comment]
run integration tests

**Comment by sijie** [comment]
@hugomiguelabreu can you rebase this change to latest master?

**Comment by hugomiguelabreu** [comment]
> @hugomiguelabreu can you rebase this change to latest master?

Yes. I rebased the feature without any conflicts. :)

**Comment by hugomiguelabreu** [comment]
run integration tests

**Comment by sijie** [comment]
@hugomiguelabreu can you rebased this to latest apache master? it seems that this pull request shows additional 22 commits.

**Comment by hugomiguelabreu** [comment]
> @hugomiguelabreu can you rebased this to latest apache master? it seems that this pull request shows additional 22 commits.

Forgot to merge into the master, only rebased the branch and pushed. Thanks for the heads up.

**Comment by hugomiguelabreu** [comment]
`run integration tests`

**Pull request merged**